frame/support/procedural/src/construct_runtime/expand/origin.rs

```diff
@@ -138,9 +145,20 @@ pub fn expand_outer_origin(
 			fn add_filter(&mut self, filter: impl Fn(&Self::Call) -> bool + 'static) {
 				let f = self.filter.clone();
 
-				self.filter = #scrate::sp_std::rc::Rc::new(Box::new(move |call| {
+				self.filter = #scrate::sp_std::rc::Rc::new(move |call| {
 					f(call) && filter(call)
-				}));
+				});
+			}
+
+			fn add_named_filter<Pallet: #scrate::traits::PalletInfoAccess, F: Fn(&Self::Call) -> bool + 'static>(
+				&mut self,
+				name: &'static str,
+				filter: F,
+			) {
+				self.named_filters.insert(
+					(<Pallet as #scrate::traits::PalletInfoAccess>::name(), name),
```

Comment:
Does this work if there are multiple instances of the same pallet?

Reply:
Yes, each instance has its own name in the runtime